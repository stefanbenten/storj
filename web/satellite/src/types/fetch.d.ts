// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

declare type Answer = {
	isSuccess: boolean;
	error: {
		code: any;
		message: any;
	};
}
