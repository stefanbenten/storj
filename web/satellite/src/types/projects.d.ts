// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

// Project is a type, used for creating new project in backend
declare type Project = {
	id: string,

	name: string,
	description: string,
	isTermsAccepted: boolean,
	createdAt: string,

	isSelected: boolean,
}

// UpdateProjectModel is a type, used for updating project description
declare type UpdateProjectModel = {
	id: string,
	description: string,
}

// TeamMemberModel stores needed info about user info to show it on UI
declare type TeamMemberModel = {
	user: {
		id: string,
		email: string,
		firstName: string,
		lastName: string,
	}
	joinedAt: string,
}
