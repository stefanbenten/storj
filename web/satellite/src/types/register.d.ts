// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

declare type RegisterData = {
	firstName: string,
	firstNameError: string,
	lastName: string,
	email: string,
	emailError: string,
	password: string,
	passwordError: string,
	repeatedPassword: string,
	repeatedPasswordError: string,
	isTermsAccepted: boolean,
	isTermsAcceptedError: boolean,
}
