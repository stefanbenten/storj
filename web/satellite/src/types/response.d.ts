// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

declare type RequestResponse<T> = {
    isSuccess: boolean,
    errorMessage: string,
    data: T
}
