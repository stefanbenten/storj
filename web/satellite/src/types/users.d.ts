// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

declare type User = {
	firstName: string,
	lastName: string,
	email: string
}

// Used in users module to pass parameters to action
declare type UpdatePasswordModel = {
    oldPassword: string,
	newPassword: string
}
