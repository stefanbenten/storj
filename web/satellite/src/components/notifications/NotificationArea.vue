// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div id="notificationArea" class="notification-container" v-if="currentNotification" >
        <Notification :type="currentNotification.type" :message="currentNotification.message" />
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import Notification from '@/components/notifications/Notification.vue';

@Component({
    computed: {
        // Computed value for current notification depends on store.state value
        currentNotification: function (): Notification {
            return this.$store.getters.currentNotification;
        }
    },
    components: {
        Notification,
    }
})

export default class NotificationArea extends Vue {
}
</script>

<style scoped lang="scss">
    .notification-container {
        height: 98px;
        max-width: 80%;
        width: 100%;
        background-color: #fff;
        display: flex;
        position: fixed;
        bottom: 50px;
        right: 30px;
        align-items: center;
        justify-content: space-between;
        box-shadow: 0px 12px 24px rgba(175, 183, 193, 0.4);
        border-radius: 6px;
        z-index: 9999;
    }
</style>
