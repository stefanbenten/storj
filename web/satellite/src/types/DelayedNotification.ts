// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

import { getId } from '@/utils/idGenerator';

export class DelayedNotification {
	public type: string;
	public message: string;
	public id: string;
	private timerId: any = null;
	private startTime: any;
	private remainingTime: any;
	private callback: Function;

    constructor(callback: Function, type: string, message: string) {
        this.callback = callback;
        this.type = type;
        this.message = message;
        this.id = getId();
        this.remainingTime = 3000;
        this.start();
    }

	public pause(): void {
		clearTimeout(this.timerId);
		this.remainingTime -= new Date().getMilliseconds() - this.startTime;
	}

	public start(): void {
		this.startTime = new Date().getMilliseconds();
		this.timerId = setTimeout(this.callback, this.remainingTime);
	}
}
