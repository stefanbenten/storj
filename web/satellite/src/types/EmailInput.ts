// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

export class EmailInput {
    public value: string;
    public error: boolean;

    constructor() {
        this.value = '';
        this.error = false;
    }

    public setError(error: boolean) {
        this.error = error;
    }
}
