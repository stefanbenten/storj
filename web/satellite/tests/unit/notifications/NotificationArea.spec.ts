// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

import { shallowMount, mount } from '@vue/test-utils';
import NotificationArea from '@/components/notifications/NotificationArea.vue';
import { NOTIFICATION_TYPES } from '@/utils/constants/notification';
import { DelayedNotification } from '@/types/DelayedNotification';

describe('Notification.vue', () => {

	it('renders correctly', () => {
		const wrapper = shallowMount(NotificationArea, {
			computed: {
				currentNotification: jest.fn(),
			},
		});

		expect(wrapper).toMatchSnapshot();
	});

	it('renders correctly with notification', () => {
		const testMessage = 'testMessage';
		const notification = new DelayedNotification(
			jest.fn(),
			NOTIFICATION_TYPES.SUCCESS,
			testMessage
		);

		const wrapper = mount(NotificationArea, {
			computed: {
				currentNotification: () => notification,
			}
		});

		expect(wrapper).toMatchSnapshot();
	});
});
