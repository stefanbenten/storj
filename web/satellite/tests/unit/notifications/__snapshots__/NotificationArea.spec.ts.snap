// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification.vue renders correctly 1`] = ``;

exports[`Notification.vue renders correctly with notification 1`] = `
<div id="notificationArea" class="notification-container">
  <div class="notification-wrap" style="background-color: rgba(214, 235, 208, 0.4);">
    <div class="notification-wrap__text">
      <div><svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
          <rect width="40" height="40" rx="10" fill="#95D486"></rect>
          <path d="M27.7625 17.0601C28.348 16.4741 28.3475 15.5243 27.7614 14.9388C27.1753 14.3533 26.2256 14.3538 25.6401 14.9399L27.7625 17.0601ZM17.7003 25.01L16.6396 26.0707C16.921 26.3521 17.3027 26.5101 17.7007 26.51C18.0986 26.5099 18.4802 26.3517 18.7615 26.0701L17.7003 25.01ZM16.0607 21.249C15.4749 20.6633 14.5251 20.6633 13.9393 21.249C13.3536 21.8348 13.3536 22.7846 13.9393 23.3704L16.0607 21.249ZM25.6401 14.9399L16.6391 23.9499L18.7615 26.0701L27.7625 17.0601L25.6401 14.9399ZM18.761 23.9493L16.0607 21.249L13.9393 23.3704L16.6396 26.0707L18.761 23.9493Z" fill="white"></path>
        </svg>
      </div>
      <p>testMessage</p>
    </div>
    <div class="notification-wrap__buttons-group"><span class="notification-wrap__buttons-group__close"><svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M23.7071 9.70711C24.0976 9.31658 24.0976 8.68342 23.7071 8.29289C23.3166 7.90237 22.6834 7.90237 22.2929 8.29289L23.7071 9.70711ZM8.29289 22.2929C7.90237 22.6834 7.90237 23.3166 8.29289 23.7071C8.68342 24.0976 9.31658 24.0976 9.70711 23.7071L8.29289 22.2929ZM9.70711 8.29289C9.31658 7.90237 8.68342 7.90237 8.29289 8.29289C7.90237 8.68342 7.90237 9.31658 8.29289 9.70711L9.70711 8.29289ZM22.2929 23.7071C22.6834 24.0976 23.3166 24.0976 23.7071 23.7071C24.0976 23.3166 24.0976 22.6834 23.7071 22.2929L22.2929 23.7071ZM22.2929 8.29289L8.29289 22.2929L9.70711 23.7071L23.7071 9.70711L22.2929 8.29289ZM8.29289 9.70711L22.2929 23.7071L23.7071 22.2929L9.70711 8.29289L8.29289 9.70711Z" fill="#354049"></path>
    </svg>
    </span></div>
  </div>
</div>
`;
